aries/aries_vcx_core/src/anoncreds/credx_anoncreds.rs

```diff
@@ -60,131 +59,47 @@ pub struct RevocationRegistryInfo {
     pub used_ids: HashSet<u32>,
 }
 
-/// Adapter used so that credx does not depend strictly on the vdrtools-wallet
-/// Will get removed when the wallet and anoncreds interfaces are de-coupled.
 #[derive(Debug)]
 struct WalletAdapter(Arc<dyn BaseWallet>);
 
 #[async_trait]
-impl BaseWallet for WalletAdapter {
-    #[cfg(feature = "vdrtools_wallet")]
-    fn get_wallet_handle(&self) -> indy_api_types::WalletHandle {
-        self.0.get_wallet_handle()
+impl RecordWallet for WalletAdapter {
+    async fn add_record(&self, record: Record) -> VcxCoreResult<()> {
+        self.0.add_record(record).await
     }
 
-    async fn create_and_store_my_did(
-        &self,
-        seed: Option<&str>,
-        kdf_method_name: Option<&str>,
-    ) -> VcxCoreResult<(String, String)> {
-        self.0.create_and_store_my_did(seed, kdf_method_name).await
-    }
-
-    async fn key_for_local_did(&self, did: &str) -> VcxCoreResult<String> {
-        self.0.key_for_local_did(did).await
-    }
-
-    async fn replace_did_keys_start(&self, target_did: &str) -> VcxCoreResult<String> {
-        self.0.replace_did_keys_start(target_did).await
-    }
-
-    async fn replace_did_keys_apply(&self, target_did: &str) -> VcxCoreResult<()> {
-        self.0.replace_did_keys_apply(target_did).await
-    }
-
-    async fn add_wallet_record(
-        &self,
-        xtype: &str,
-        id: &str,
-        value: &str,
-        tags: Option<HashMap<String, String>>,
-    ) -> VcxCoreResult<()> {
-        self.0.add_wallet_record(xtype, id, value, tags).await
-    }
-
-    async fn get_wallet_record(
-        &self,
-        xtype: &str,
-        id: &str,
-        options: &str,
-    ) -> VcxCoreResult<String> {
-        self.0.get_wallet_record(xtype, id, options).await
-    }
-
-    async fn get_wallet_record_value(&self, xtype: &str, id: &str) -> VcxCoreResult<String> {
-        self.0.get_wallet_record_value(xtype, id).await
-    }
-
-    async fn delete_wallet_record(&self, xtype: &str, id: &str) -> VcxCoreResult<()> {
-        self.0.delete_wallet_record(xtype, id).await
-    }
-
-    async fn update_wallet_record_value(
-        &self,
-        xtype: &str,
-        id: &str,
-        value: &str,
-    ) -> VcxCoreResult<()> {
-        self.0.update_wallet_record_value(xtype, id, value).await
-    }
-
-    async fn add_wallet_record_tags(
-        &self,
-        xtype: &str,
-        id: &str,
-        tags: HashMap<String, String>,
-    ) -> VcxCoreResult<()> {
-        self.0.add_wallet_record_tags(xtype, id, tags).await
+    async fn get_record(&self, name: &str, category: &str) -> VcxCoreResult<Record> {
+        self.0.get_record(name, category).await
     }
 
-    async fn update_wallet_record_tags(
+    async fn update_record_tags(
         &self,
-        xtype: &str,
-        id: &str,
-        tags: HashMap<String, String>,
+        name: &str,
+        category: &str,
```

Comment:
reviewer's note: the order has been swapped
- `name` is equivalent to former `id`
- `category` is equivalent to former `xtype`

Is there rationale for the changed order? I have actually slight preference for the original order (and would reduce PR delta) but it's not a biggie. Personal preference, feel free to mark this resolved. (or if you would be changing this, note that InteliJ has great refactoring feature for changing function arguments order)